From 6c53fbcdfd642d2dd5fb1c624642bc54426e8e97 Mon Sep 17 00:00:00 2001
From: Erik Chen <erikchen@chromium.org>
Date: Tue, 20 Aug 2024 07:14:42 +0000
Subject: [PATCH] Avoid crash in SecurityStatePageLoadMetricsObserver.

SecurityStatePageLoadMetricsObserver can be created in situations
without SecurityStateTabHelper. While this may not conceptually makes
sense, this currently happens in practice for
SideSearchTabContentsHelper::CreateSidePanelContents.

Bug: 355894536
Change-Id: Ibe9d218a5d072c566c2dade20e6823b2c3478254
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/5799315
Reviewed-by: Thomas Lukaszewicz <tluk@chromium.org>
Commit-Queue: Erik Chen <erikchen@chromium.org>
Code-Coverage: findit-for-me@appspot.gserviceaccount.com <findit-for-me@appspot.gserviceaccount.com>
Cr-Commit-Position: refs/heads/main@{#1343978}
---

--- a/chrome/browser/page_load_metrics/observers/security_state_page_load_metrics_observer.cc
+++ b/chrome/browser/page_load_metrics/observers/security_state_page_load_metrics_observer.cc
@@ -228,6 +228,13 @@
   // resolved.
   security_state_tab_helper_ =
       SecurityStateTabHelper::FromWebContents(web_contents);
+  // TODO(https://crbug.com/355894536): There are some features that currently
+  // instantiate a SecurityStatePageLoadMetricsObserver without a
+  // ChromeSecurityStateTabHelper. This does not make sense conceptually. For
+  // now add an early return.
+  if (!security_state_tab_helper_) {
+    return;
+  }
 
   DCHECK_EQ(initial_security_level_, security_state::NONE);
   DCHECK_EQ(current_security_level_, security_state::NONE);
